### PR TITLE
GEODE-9994 Make Redis RENAME atomic

### PR DESCRIPTION
**Pull request by ezoerner**
Most of the work here was on getting a DUnit test to fail with the old version of the code. The test doesn't fail every time when run with the old code, but it does fail sometimes. With the new code, i.e. the fix, I haven't seen the test fail at all.

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/key/RenameDUnitTest.java [hunk 1]
```diff
@@ -210,6 +227,48 @@ private void doConcurrentRenames(List<String> listOfKeys1, List<String> listOfKe
...
+    final Future<Void> future1 = executor.runAsync(task1);
+    final Future<Void> future2 = executor.runAsync(task2);
+    final Future<Void> future3 = executor.runAsync(task3);
+
+    final String finalRedisPort = Integer.toString(server3Port);
+    final int finalLocatorPort = locatorPort;
+    final Future<?> crasherFuture = executor.submit(() -> {
+      try {
+        for (int i = 0; i < 20 && running.get(); i++) {
+          clusterStartUp.moveBucketForKey(hashtags.get(0), "server-3");
```
Comment (DonalEvans) [inline geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/key/RenameDUnitTest.java:252]: Is this necessary? It seems like crashing the server should be sufficient to trigger the bug, without needing to move the bucket first.

Comment (ezoerner) [reply geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/key/RenameDUnitTest.java:252]: This follows the pattern in MSetDUnitTest `testMSet_crashDoesNotLeaveInconsistencies` which Jens wrote. I believe the purpose of this line to make sure there are keys hosted on the server that is about to be crashed.

**Comment by ezoerner** [comment]
Hmm, just after I submitted this PR I ran the test again locally and it failed because the new key existed even though the RENAME failed.  I will have to take a closer look at this to make sure it's not a logic error in the test.

**Comment by ezoerner** [comment]
The build failure here appears to be unrelated (a groovy error?)

File: geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/key/RenameDUnitTest.java (hunk 1 above)
Comment (ezoerner) [reply geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/key/RenameDUnitTest.java:252]: Perhaps when the server has previously crashed there may not be keys hosted there anymore. Maybe Jens can comment on this as well if there's more to it than this.

**Comment by ezoerner** [comment]
Also, the error messages in the DUnit test need to be improved so they reflect the boolean logic correctly

**Review by dschneider-pivotal — APPROVED** [review]

**Comment by ezoerner** [comment]
I've modified the verification to allow for erroneous client reties on successful renames that appear to be errors

**Comment by ezoerner** [comment]
Just waiting on all the checks to complete, and in the meantime I am re-running the current version of the test with the old non-transactional code to try and make it fail. Regardless though, once the checks pass I think this can be merged.

**Review by onichols-pivotal — CHANGES_REQUESTED** [review]
First commit (and also PR title) must begin with `GEODE-nnnnn: `.  Please amend your initial commit and force-push.

**Review by onichols-pivotal — CHANGES_REQUESTED** [review]
First commit (and also PR title) must begin with `GEODE-nnnnn: `.  Please amend your initial commit and force-push.

**Pull request merged**